modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
-import java.net.URL;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.util.*;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import static java.net.HttpURLConnection.HTTP_OK;
-import static org.testcontainers.couchbase.CouchbaseContainer.CouchbasePort.*;
 
 /**
- * Based on Laurent Doguin version,
+ * The couchbase container initializes and configures a Couchbase Server single node cluster.
  * <p>
- * optimized by Tayeb Chlyah
+ * Note that it does not depend on a specific couchbase SDK, so it can be used with both the Java SDK 2 and 3 as well
+ * as the Scala SDK 1 or newer. We recommend using the latest and greatest SDKs for the best experience.
  */
-@AllArgsConstructor
 public class CouchbaseContainer extends GenericContainer<CouchbaseContainer> {
 
-    public static final String VERSION = "5.5.1";
-    public static final String DOCKER_IMAGE_NAME = "couchbase/server:";
-    public static final ObjectMapper MAPPER = new ObjectMapper();
-    public static final String STATIC_CONFIG = "/opt/couchbase/etc/couchbase/static_config";
-    public static final String CAPI_CONFIG = "/opt/couchbase/etc/couchdb/default.d/capi.ini";
+    private static final int MGMT_PORT = 8091;
+
+    private static final int MGMT_SSL_PORT = 18091;
+
+    private static final int VIEW_PORT = 8092;
 
-    private static final int REQUIRED_DEFAULT_PASSWORD_LENGTH = 6;
+    private static final int VIEW_SSL_PORT = 18092;
 
-    private String memoryQuota = "300";
+    private static final int QUERY_PORT = 8093;
 
-    private String indexMemoryQuota = "300";
+    private static final int QUERY_SSL_PORT = 18093;
 
-    private String clusterUsername = "Administrator";
+    private static final int SEARCH_PORT = 8094;
 
-    private String clusterPassword = "password";
+    private static final int SEARCH_SSL_PORT = 18094;
 
-    private boolean keyValue = true;
+    private static final int KV_PORT = 11210;
 
-    @Getter
-    private boolean query = true;
+    private static final int KV_SSL_PORT = 11207;
 
-    @Getter
-    private boolean index = true;
+    private static final String DOCKER_IMAGE_NAME = "couchbase/server";
 
-    @Getter
-    private boolean primaryIndex = true;
+    private static final String VERSION = "6.5.0";
 
-    @Getter
-    private boolean fts = false;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    @Getter(lazy = true)
-    private final CouchbaseEnvironment couchbaseEnvironment = createCouchbaseEnvironment();
+    private static final OkHttpClient HTTP_CLIENT = new OkHttpClient()
+        .newBuilder()
+        .readTimeout(Duration.ofMinutes(1))
```

Comment:
Big timeout necessary for startup?

Reply:
there are per-strategy timeouts too, I see no harm here

Reply:
We can reduce it too if needed, but I figured maybe on slower machines it might help not break